src/python/pim/polynomial_representation.py

```diff
@@ -0,0 +1,89 @@
+from typing import List
+import re
+import itertools as it
+import numpy as np
+
+class Polynomial:
+    def __init__(self, coefficients: List[int]) -> None:
+        self.coefficients: List[int] = coefficients
+
+    def __str__(self) -> str:
+        """returns a string representation of the polynomial"""
+        coeff_list: List[str] = [display(c, i) for i, c in enumerate(self.coefficients)]
+        filter_coeff_list = list(filter(lambda x: x != "", coeff_list))
+        return re.sub(r"\+ -", "- ", " + ".join(filter_coeff_list))
+
+    def __add__(self, other: "Polynomial") -> "Polynomial":
+        """returns the sum of two polynomials (like degrees are added together)"""
+        return [sum(n) for n in it.zip_longest(self.coefficients, other.coefficients, fillvalue = 0)]
+
+    def __mul__(self, other: "Polynomial") -> "Polynomial":
+        """returns the product of two polynomials"""
+        len_self: int = len(self.coefficients)
+        len_other: int = len(other.coefficients)
+        blank_coeff_list: list(int) = [0] * (len_self + len_other - 1)
+
+        #Multiplies two polynomials term by term
+        for i in range(len_self):
+            for j in range(len_other):
+                blank_coeff_list[i + j] += self.coefficients[i] * other.coefficients[j]
+        return blank_coeff_list
+
+    def evaluate(self, value: int) -> int:
+        """evaluates a polynomial at a given value"""
+        calculate_list: list[str] = [calculate(c, i) for i, c in enumerate(self.coefficients)]
```

Comment:
First, just like with `display()`, I suspect that `calculate()` is a function that only relates to instances of `Polynomial`. As such, I would recommend making it static.

Second, `calculate_list` seems like it should be of type `List[Tuple[int, int]]` based upon the function signature of `calculate()`. Is that correct?

Reply:
I added calculate as a staticmethod to the class.

I also fixed the output list to be a List[Polyterm] ....more on that later.